src/parseDef.ts

```diff
@@ -94,14 +94,17 @@ export function parseDef(
   refs.seen.set(def, newItem);
 
   const jsonSchema = selectParser(def, (def as any).typeName, refs);
+  const fallbackJsonSchema = refs.onParseDef?.(def, refs, jsonSchema);
 
-  if (jsonSchema) {
-    addMeta(def, refs, jsonSchema);
+  const resolvedJsonSchema = jsonSchema ?? fallbackJsonSchema ?? undefined;
```

Comment:
I think the priority is weird here. As a user I would expect my output to have priority.

Also maybe call it manualSchema?

Reply:
That's a good point. I will change priority here and rename.